### PR TITLE
Allow changing forceRefresh

### PR DESCRIPTION
**Pull request by jeresig**
Re-implementation of:
https://github.com/ReactTraining/history/pull/614

Includes the built artifacts so that we can use it in from `package.json`.

Issue: https://github.com/ReactTraining/history/pull/614

Test plan:
It should now be possible to call `setForceRefresh` and dynamically change this setting!

I tested this in Webapp by having two browser windows open side-by-side, both logged in as the same user. I then logged out of one and tabbed back to the other. I saw a notice in the console that a Force Refresh was triggered. I think navigated in the Course Menu and clicked "Early Math" and had a full page refresh, it didn't attempt to use client-side nav!

### CODE REVIEW

**Comment by k4b7** [comment]
> I think it'd be good to have tests for this although I'm not sure if it would be easier to write a test here or have a test for the component you're writing in webapp that uses this functionality.

In webapp, I think. That way if we go back to the canonical library, we aren't diverged too greatly.

**Comment by somewhatabstract** [comment]
Can you add the build artifacts to .gitattributes so they don't appear in diffs?

File: .gitattributes [hunk 1]
```diff
@@ -0,0 +1 @@
+esm/*.js binary
```
Comment (somewhatabstract) [inline .gitattributes:1]: Hmm, this doesn't appear to have worked for this diff in the mobile app. I think this is ok, but I would've used the `diff` modifier, I think (that's what I used in render-gateway).

**Review by somewhatabstract — APPROVED** [review]
Seems fine to me.

File: .gitattributes (hunk 1 above)
Comment (jeresig) [reply .gitattributes:1]: In `webapp` we use `binary`, as far as I can tell.

**Comment by k4b7** [comment]
> I then logged out of one and tabbed back to the other. I saw a notice in the console that a Force Refresh was triggered. I think navigated in the Course Menu and clicked "Early Math" and had a full page refresh, it didn't attempt to use client-side nav!

Navigating to a course via the course menu is a client-side nav on prod.  Does a user have to navigate for the force refresh to actually do the full page reload?

**Comment by jeresig** [comment]
@kevinbarabash There are many ways in which we trigger the force refresh (automatic of manual). In this case we trigger it my causing the page to reload when the user navigates on the client. See the last bullet point inside "Decision" in ADR #301: https://docs.google.com/document/d/1FstRfSQ90iadq7ywGRPksS89AO9asxuKd-N3nDnW2zg/edit#heading=h.y789xpoo4j6h

**Pull request merged**